src/components/Category.tsx

```diff
@@ -75,34 +84,100 @@ const Category = ({ category }: Props) => {
     setTasks(newTasks)
   }
 
+  const catEditRef = useRef<HTMLInputElement>(null)
+
+  const handleBlur = (event: React.FocusEvent<HTMLDivElement>) => {
+    // Guard for Ok and delete buttons
+    if (event.currentTarget.contains(event.relatedTarget)) return;
+
+    handleCancel();
+  };
+
+  function handleOk() {
+    console.log("handling Ok")
+    const newName = catEditRef.current?.value;
+    if (!newName) return;
+
+    const updatedCategory = {
+      ...category,
+      name: newName
+    };
+    onUpdateCategory(updatedCategory);
+    setIsCategoryEditVisible(false);
```

Comment:
could probably set this to handleCategoryPopup but this works fine too. I think I only did that for tasks because I didn't add a way to pass the useState down specifically